File: src/main/java/net/zomis/chess/ChessPosition.java
package net.zomis.chess;

public class ChessPosition {

    private static final int SIZE = 8;

    private final int x;
    private final int y;

    public ChessPosition(int x, int y) {
        this.x = x;
        this.y = y;
    }

    public static ChessPosition fromMoveNotation(String move) {
        move = move.toUpperCase();
        char col = move.charAt(move.length() - 2);
        int row = Character.digit(move.charAt(move.length() - 1), 10);
        return new ChessPosition(col - 'A', row - 1);
    }

    public char getColumn() {
        return (char) ('A' + x);
    }

    public ChessPosition offset(int x, int y) {
        return new ChessPosition(this.x + x, this.y + y);
    }

    public int getY() {
        return y;
    }

    public int getX() {
        return x;
    }

    @Override
    public String toString() {
        return String.valueOf(getColumn()) + getRow();
    }

    public int getRow() {
        return getY() + 1;
    }

    public boolean onBoard() {
        return x >= 0 && x < SIZE && y >= 0 && y < SIZE;
    }
}

File: chess-core/src/main/kotlin/net/zomis/chess/core/ChessPosition.kt
package net.zomis.chess.core

class ChessPosition(val x: Int, val y: Int) {

    val column: Char
        get() = ('A' + x).toChar()

    val row: Int
        get() = y + 1

    fun offset(x: Int, y: Int): ChessPosition {
        return ChessPosition(this.x + x, this.y + y)
    }

    override fun toString(): String {
        return column.toString() + row
    }

    fun onBoard(): Boolean {
        return x >= 0 && x < SIZE && y >= 0 && y < SIZE
    }

    companion object {
        private val SIZE = 8

        fun fromMoveNotation(move: String): ChessPosition {
            var move = move
            move = move.toUpperCase()
            val col = move[move.length - 2]
            val row = move[move.length - 1].toInt()
            return ChessPosition(col - 'A', row - 1)
        }
    }
}
